src/event/data.rs

```diff
@@ -82,9 +82,40 @@ impl TryFrom<Data> for String {
 impl fmt::Display for Data {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
         match self {
-            Data::Binary(vec) => write!(f, "Binary data: {:?}", str::from_utf8(vec).unwrap()),
+            Data::Binary(vec) => {
+                write!(f, "Binary data: ")?;
+                let mut slice = &vec[..];
+                loop {
+                    match str::from_utf8(slice) {
+                        Ok(s) => break f.write_str(s),
+                        Err(e) => {
+                            let (good, bad) = slice.split_at(e.valid_up_to());
+
+                            // SAFETY: good is a valid utf8 sequency
+                            f.write_str(unsafe { str::from_utf8_unchecked(good) })?;
```

Comment:
Is there a way we can remove this unsafe?
I'm afraid this adds some security problems.